crypto/src/e2e_identity/identity.rs

```diff
@@ -156,20 +156,18 @@ impl MlsConversation {
             return Err(CryptoError::ConsumerError);
         }
         let user_ids = user_ids.iter().map(|uid| uid.as_bytes()).collect::<Vec<_>>();
+
         self.members_with_key()
             .iter()
             .filter_map(|(id, c)| UserId::try_from(id.as_slice()).ok().zip(Some(c)))
             .filter(|(uid, _)| user_ids.contains(uid))
             .map(|(uid, c)| (uid, c.extract_identity(self.ciphersuite(), env)))
-            .chunk_by(|(uid, _)| *uid)
-            .into_iter()
-            .map(|(uid, group)| {
+            .map(|(uid, identity)| {
                 let uid = String::try_from(uid);
-                let identities = group.into_iter().map(|(_, id)| id).collect::<CryptoResult<Vec<_>>>();
                 // could be simplified if `Result::zip` was available
-                uid.and_then(|uid| identities.map(|ids| (uid, ids)))
+                uid.and_then(|uid| identity.map(|id| (uid, id)))
             })
-            .collect::<CryptoResult<HashMap<_, _>>>()
+            .process_results(|iter| iter.into_group_map())
     }
 }
```

Comment:
Could you please factor this out into a separate commit, so that it is not mixed in with a bunch of unrelated changes? It would be good to mention in the commit message what was wrong with previous code.